requre/online_replacing.py

```diff
@@ -267,39 +288,42 @@ def replace_module_match(
     def decorator_cover(func):
         @functools.wraps(func)
         def _internal(*args, **kwargs):
-            original_key_strategy = DataMiner().key_stategy_cls
-            # use simple strategy, to not store stack info as keys
-            DataMiner().key_stategy_cls = storage_keys_strategy
-            original_storage_file = _change_storage_file(
-                storage_file=storage_file, func=func, args=args
-            )
+            if cassette:
+                cassette_int = cassette
+            else:
+                cassette_int = Cassette()
+            # set storage if if not set to default one, based on function name
+            if cassette_int.storage_file is None:
+                _change_storage_file(cassette=cassette_int, func=func, args=args)
+            cassette_int.data_miner.key_stategy_cls = storage_keys_strategy
             # ensure that directory structure exists already
-            os.makedirs(
-                os.path.dirname(PersistentObjectStorage().storage_file), exist_ok=True
-            )
+            os.makedirs(os.path.dirname(cassette_int.storage_file), exist_ok=True)
             # Store values and their replacements for modules to be able to revert changes back
             original_module_items = _parse_and_replace_sys_modules(
-                what=what, decorate=decorate, replace=replace
+                what=what, cassette=cassette_int, decorate=decorate, replace=replace
             )
             try:
+                # pass current cassette to underneath decorator and do not overwrite if set there
+                if (
+                    "cassette" in inspect.getfullargspec(func).annotations
+                    and inspect.getfullargspec(func).annotations["cassette"] == Cassette
+                    and "cassette" not in kwargs
```

Comment:
Will this work also for more than two decorators? (I am just wondering if the `cassette` argument will not be in `kwargs` when we forward that.)

Reply:
in case it is already in kwargs, then it should not be replaced, or better to say, there is question how it should behave, example:
```
@replace(what="neco", storage_file="x")
@replace(what="ostatni", storage_file="y")
def test(self)
...
```

what is proper behavior? should that be possible to store more files or raised error, or the highest decorator should replace all underneath cassettes? this code tries to solve, that in case cassette is not placed it is not replaced. But maybe it is highly possible that I have there some mistake, would be nice to test it extensively if the behaviour is exacly what I think it should do :-)

Reply:
For your example, I (from the user perspective) would save `neco` responses to `x` and `ostatni` responses to `y`. Is it doable? Or, raise an error and support only this:

```python
@replace(what="neco", storage_file="x")
@replace(what="ostatni")
def test(self)
...
```

Or

```python
@set_cassette_file("x")
@replace(what="neco")
@replace(what="ostatni")
def test(self)
...
```